docs/contracts.rst

```diff
@@ -1035,8 +1034,7 @@ more advanced example to implement a set).
             // The library functions can be called without a
             // specific instance of the library, since the
             // "instance" will be the current contract.
-            if (!Set.insert(knownValues, value))
-                throw;
+            assert(Set.insert(knownValues, value));
```

Comment:
I think even here, we should use `require` because it is not an internal error if this fails - it is still invalid usage of contract `C`.